src/x11/x11-window.c

```diff
@@ -1611,15 +1611,15 @@ static EGLBoolean SyncRendering(EplDisplay *pdpy, EplSurface *surf, X11ColorBuff
         return EGL_TRUE;
     }
 
-    pwin->inst->platform->priv->egl.Flush();
-
     sync = pwin->inst->platform->priv->egl.CreateSync(pwin->inst->internal_display->edpy,
             EGL_SYNC_NATIVE_FENCE_ANDROID, NULL);
     if (sync == EGL_NO_SYNC)
     {
         goto done;
     }
 
+    pwin->inst->platform->priv->egl.Flush();
```

Comment:
Despite what the spec claims, this isn't actually how our implementation behaves. It flushes internally during CreateSync() and attaches the fence created by that flush operation. I think this was and is still just perturbing some timing/race if it affects anything at all.

Reply:
That's possible -- I'm still trying to make sense of that myself.

But at the very least, this is the one sequence that the spec requires to work, so even if there's a driver bug lurking, this should continue to work once it's fixed, right?

Reply:
In that it won't make things worse, yes. I just worry there's a worse bug somewhere that this continues to paper over. This might actually break the papering over, in that the after-create flush will probably get reduced to a no-op by the driver logic.